### PR TITLE
Move all the types to separate interface file.

### LINKED ISSUE #75: Move out all the typescript types in single file.
Currently, we have typescript types in all the files, we should move out those and keep them in a single file and use it from there.
Also, instead of `type` make it `interface`.

### PR DESCRIPTION
**Pull request by ShaileshParmar11**
Move the all the types to its separate interface file for better readability and re-usability.
Closes #75 

@ashutosh1919 @Sachin-chaurasiya

### CODE REVIEW

File: src/lib/shapes/shapes.interface.ts [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+/**
+ * This interface is used for the shape which required additional props along with ShapeProps
+ */
+export interface ShapeWithSize extends ShapeProps, SizeProps {}
+
+export interface BaseShapeProps extends CommonProps, SizeProps {
+  children: ReactNode;
+}
+
+export interface DonutProps extends ShapeProps {
```
Comment (Sachin-chaurasiya) [inline src/lib/shapes/shapes.interface.ts:39]:
```suggestion
export interface DonutProps extends ShapeProps,Pick<SizeProps,"width"> {}
```
@ShaileshParmar11 Here you can use Typescript util **Pick** to make DonutProps.

**Review by Sachin-chaurasiya — CHANGES_REQUESTED** [review]
One change is required, Otherwise LGTM!
@ashutosh1919 Please review once.

File: src/lib/shapes/shapes.interface.ts (hunk 1 above)
Comment (ShaileshParmar11) [reply src/lib/shapes/shapes.interface.ts]: Hey, great suggestion, code updated accordingly.

**Review by Sachin-chaurasiya — APPROVED** [review]

**Comment by Sachin-chaurasiya** [comment]
@ShaileshParmar11, Please rebase your branch with main.

**Review by Sachin-chaurasiya — APPROVED** [review]

**Review by ashutosh1919 — APPROVED** [review]
LGTM! @ShaileshParmar11 Thanks for contribution.

**Pull request merged**